cirq/ops/clifford_gate.py

```diff
@@ -207,8 +209,25 @@ def _value_equality_values_(self):
                 self.transform(pauli_gates.Z))
 
     def __pow__(self, exponent) -> 'SingleQubitCliffordGate':
-        if exponent != -1:
+        if exponent == 0.5:
```

Comment:
Instead of a big if/else block, I would define a private module-level constant dictionary mapping (float, gate) to gate, with each of the special cases encoded.